package-scripts/datadog-agent/preinst

```diff
@@ -42,7 +42,15 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
     if $( apt-key finger | grep -q 'A292 3DFF 56ED A6E7 6E55  E492 D3A8 0E30 382E 94DE' ); then
         echo "... key already installed"
     else
-        apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 382E94DE >/dev/null 2>&1 && echo " ... OK" || echo " ... failed"
+        apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 382E94DE >/dev/null 2>&1
+        if [ $? -eq 0 ]; then
+            echo " ... OK"
+        else
+            echo " ... failed"
+            if [ -f "/etc/debian_version" ] || [ "$DISTRIBUTION" = "Debian" ]; then
+                 echo "The dirmngr package is needed to add a new key to APT on Debian since version 9"
```

Comment:
What do you think about adding more details here on the consequences of failing to add the new APT key. Something like: `WARNING: The "dirmngr" package is needed to install the new APT key required by future versions of the Agent. If this package is not installed, please install it and then re-install the agent. The agent installation may break in the future if the new APT key is not installed on your system`